### DISCUSSION TITLE
Translated Screenshots

### DISCUSSION THREAD

**Discussion by JakobDev** [ximion/appstream#471] [Q&A] [answered]
[According to the documentation](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-screenshots), I should be able to provide translated screenshots using the `xml:lang` attribute. I have the following example file:
```xml
<?xml version='1.0' encoding='utf-8'?>
<component type="desktop">
  <id>org.example.App</id>
  <name>Test</name>
  <summary>Test</summary>
  <screenshots>
    <screenshot type="default">
      <image type="source">https://upload.wikimedia.org/wikipedia/commons/c/ce/PNG_demo_Banana.png</image>
      <image type="source" xml:lang="de">https://upload.wikimedia.org/wikipedia/commons/3/39/PNG_demo_heatmap_Banana.png</image>
    </screenshot>
  </screenshots>
  <content_rating type="oars-1.1"/>
</component>
```
This is a little examples which contains just 2 images from Wikipedia for testing reasons.

If I view it with:
```
gnome-software --show-metainfo=org.example.App.metainfo.xml
```
As I have A German System, I see the second Image.

Now I try to view it in English:
```
LC_ALL=C gnome-software --show-metainfo=org.example.App.metainfo.xml
```
The User interface of Gnome Software is now in English, but I still see the second Image, which is ,marked as the German translation. Is this a Bug in Gnome Software or is just the Documentation wrong?

**Comment by ximion** [comment, marked as answer]
This is a bug in GNOME Software which apparently doesn't use libappstream to read this data and is doing things differently, apparently ignoring the locale tag entirely.
So, please file a bug against GS ;-)

**Reply by JakobDev** [reply]
[Done](https://gitlab.gnome.org/GNOME/gnome-software/-/issues/2117)

I wonder if this also happens with Disocver, but it sadly has no Way to preview a AppStream file.

**Reply by ximion** [reply]
Is the app published? I could check that way (or you can dump the metainfo file into a system location, that may also work).

**Reply by JakobDev** [reply]
This is not published. This was just for testing if it works in before actually using it in a real App. I mostly develop simple tools which have one or 2 Screenshots, so adding a translated one would not hurt. As this currently causes Bugs, I think I will not use it in the next time. I had alrdy [opened a Feature Request](https://bugs.kde.org/show_bug.cgi?id=467248) to add a Preview Function to Discover, as this is very useful to test things out.
